kernel/interrupts/src/aarch64/mod.rs

```diff
@@ -0,0 +1,416 @@
+use core::arch::global_asm;
+use core::cell::UnsafeCell;
+use core::ops::DerefMut;
+use core::fmt;
+
+use cortex_a::registers::*;
+
+use tock_registers::interfaces::Writeable;
+use tock_registers::interfaces::Readable;
+use tock_registers::registers::InMemoryRegister;
+
+use gic::{qemu_virt_addrs, ArmGic, IntNumber, Version as GicVersion};
+use irq_safety::{RwLockIrqSafe, MutexIrqSafe};
+use memory::get_kernel_mmi_ref;
+use log::{info, error};
+
+// Raw/Early exception handling is defined
+// in this assembly file.
+global_asm!(include_str!("table.s"));
+
+// The global interrupt controller singleton
+static GIC: MutexIrqSafe<Option<ArmGic>> = MutexIrqSafe::new(None);
+
+// Default Timer IRQ number on AArch64 as
+// defined by Arm Manuals
+pub const AARCH64_TIMER_IRQ: IntNumber = 30;
+
+const MAX_IRQ_NUM: usize = 256;
+
+// Singleton which acts like an x86-style
+// Interrupt Descriptor Table: it's an
+// array of function pointers which are
+// meant to handle IRQs. Synchronous
+// Exceptions (syscalls) are not IRQs on
+// aarch64; this crate doesn't expose any
+// way to handle them at the moment.
+static IRQ_HANDLERS: RwLockIrqSafe<[HandlerFunc; 256]> = RwLockIrqSafe::new([default_irq_handler; 256]);
+
+/// Wrapper structs for memory copies of registers.
+#[repr(transparent)]
+struct SpsrEL1(InMemoryRegister<u64, SPSR_EL1::Register>);
+struct EsrEL1(InMemoryRegister<u64, ESR_EL1::Register>);
```

Comment:
are both of these supposed to be transparent? the doc comment and `repr` attribute only apply to the first one right now.
You should also provide better docs for each of these structs, i.e., describe what Spsr and Esr stand for and what they're used for.

Reply:
Wow I fixed this in the past, why wasn't it commited I'm losing my mind